framework/crud_resource.go

```diff
@@ -33,11 +38,8 @@ func NewCRUDResource(config CRUDResourceConfig) (*CRUDResource, error) {
 	if config.Ops == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.Ops must not be empty")
 	}
-	if config.Ops.Name() == "" {
-		return nil, microerror.Maskf(invalidConfigError, "config.Ops.Name() must not be empty")
-	}
-	if config.Ops.Underlying() == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.Ops.Underlying() must not be empty")
+	if config.Name == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Name must not be empty")
```

Comment:
`Name` is a setting and should have its own block separated from the dependencies.

Reply:
Newline added.